sklearn/utils/validation.py

```diff
@@ -498,10 +498,10 @@ def check_X_y(X, y, accept_sparse=None, dtype="numeric", order=None, copy=False,
                     ensure_2d, allow_nd, ensure_min_samples,
                     ensure_min_features, warn_on_dtype, estimator)
     if multi_output:
-        y = check_array(y, 'csr', force_all_finite=True, ensure_2d=False,
-                        dtype=None)
+        y = check_array(y, 'csr', copy=copy, force_all_finite=True,
+                        ensure_2d=False, dtype=None)
```

Comment:
we could be really magic and copy only if ``y_numeric`` [not really a suggestion]. For classifiers, we never want to copy ``y``. We could have ``copy_y``? But there are so many arguments already!